src/modules/connection.js

```diff
@@ -0,0 +1,102 @@
+var amqp = require('amqplib'),
+  assert = require('assert'),
+  packageVersion = require('../../package.json').version,
+  startedAt = new Date();
+
+var connections = {};
+
+/**
+ * Connect to the broker. We keep only 1 connection for each connection string provided in config, as advised by RabbitMQ
+ * @return {Promise} A promise that resolve with an amqp.node connection object
+ */
+function getConnection() {
+  let url = this.config.url;
+  let hostname = this.config.hostname;
+  let connection = connections[url];
+
+  //cache handling, if connection already opened, return it
+  if (connection && connection.conn) {
+    return Promise.resolve(connection.conn);
+  }
+
+  //prepare the connection internal object, and reset channel if connection has been closed
+  connection = connections[url] = { conn: null, channel: null };
+  connection.conn = amqp.connect(url, { clientProperties:
+      { hostname: hostname, bunnymq: packageVersion, startedAt: startedAt, connectedAt: new Date() }
+    }).then((conn) => {
+        //on connection close, delete connection
+        conn.on('close', () => { delete connection.conn; });
+        conn.on('error', this.config.transport.error);
+
+        connection.conn = conn;
+        return conn;
+    })
+    .catch(() => {
+      connection.conn = null;
+    });
+
+  return connection.conn;
```

Comment:
The function docs says it returns a Promise so it should do so allways. (Unless the connection.coon has a then function). Besides you resolved it to a promise on line 19 of this file so you could do it here and not have to worry to promisify where you use it ('getConnection'). The same thing goes for getChannel.